test/holder/burn_decimal_test.go

```diff
@@ -0,0 +1,269 @@
+
+			BeforeEach(func() {
+				tx, err := TKNBurner.Mint(Owner.TransactOpts(), BankAccount.Address(), big.NewInt(583))
+				Expect(err).ToNot(HaveOccurred())
+				Backend.Commit()
+				Expect(isSuccessful(tx)).To(BeTrue())
+			})
+
+			It("should increase the total TKN supply to 583", func() {
+				s, err := TKNBurner.TotalSupply(nil)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(s.String()).To(Equal("583"))
+			})
+
+			It("should increase TKN balance of the specific address", func() {
+				b, err := TKNBurner.BalanceOf(nil, BankAccount.Address())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(b.String()).To(Equal("583"))
+			})
+
+			It("Should emit a Transfer event", func() {
+				from := []common.Address{common.HexToAddress("0x0")}
+				to := []common.Address{BankAccount.Address()}
+				it, err := TKNBurner.FilterTransfer(nil, from, to)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(it.Next()).To(BeTrue())
+				evt := it.Event
+				Expect(it.Next()).To(BeFalse())
+				Expect(evt.From).To(Equal(common.HexToAddress("0x0")))
+				Expect(evt.To).To(Equal(BankAccount.Address()))
+				Expect(evt.Value.String()).To(Equal("583"))
+			})
+
+			When("187 TKN are transfered to a random address", func() {
+
+				BeforeEach(func() {
+					tx, err := TKNBurner.Transfer(BankAccount.TransactOpts(), RandomAccount.Address(), big.NewInt(187))
+					Expect(err).ToNot(HaveOccurred())
+					Backend.Commit()
+					Expect(isSuccessful(tx)).To(BeTrue())
+				})
+
+				It("should increase TKN balance of the random address", func() {
+					b, err := TKNBurner.BalanceOf(nil, RandomAccount.Address())
+					Expect(err).ToNot(HaveOccurred())
+					Expect(b.String()).To(Equal("187"))
+				})
+
+				It("should decrease TKN balance of the initial address", func() {
+					b, err := TKNBurner.BalanceOf(nil, BankAccount.Address())
+					Expect(err).ToNot(HaveOccurred())
+					Expect(b.String()).To(Equal("396"))
+				})
+
+				It("should leave the total supply intact", func() {
+					s, err := TKNBurner.TotalSupply(nil)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(s.String()).To(Equal("583"))
+				})
+
+				It("Should emit a Transfer event", func() {
+					from := []common.Address{BankAccount.Address()}
+					to := []common.Address{RandomAccount.Address()}
+					it, err := TKNBurner.FilterTransfer(nil, from, to)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(it.Next()).To(BeTrue())
+					evt := it.Event
+					Expect(it.Next()).To(BeFalse())
+					Expect(evt.From).To(Equal(BankAccount.Address()))
+					Expect(evt.To).To(Equal(RandomAccount.Address()))
+					Expect(evt.Value.String()).To(Equal("187"))
+				})
+
+				When("The holder contract owns two types of ERC20 tokens and 1 ETH", func() {
+
+                    //add the tokens to the list
+					BeforeEach(func() {
+                        tokens := []common.Address{common.HexToAddress("0x0"), ERC20Contract1Address, ERC20Contract2Address}
```

Comment:
indentation ...

Reply:
funny thing, it must be a github thing, no indentation issue in my editor...